packages-native/intro-screen/src/SwipeableContainer.tsx

```diff
@@ -0,0 +1,310 @@
+
+const isIphoneWithNotch = Platform.OS === "ios" && DeviceInfo.hasNotch();
+
+const isAndroidRTL = I18nManager.isRTL && Platform.OS === "android";
+
+interface SwipeableContainerProps {
+    skipLabel: string;
+    doneLabel: string;
+    nextLabel: string;
+    prevLabel: string;
+    showDoneButton: boolean;
+    showSkipButton: boolean;
+    showNextButton: boolean;
+    showPrevButton: boolean;
+    onSlideChange: (next: number, previous: number) => void;
+    bottomButton: boolean;
+    numberOfButtons: number;
+    renderDoneButton: () => ReactNode;
+    renderSkipButton: () => ReactNode;
+    renderPrevButton: () => ReactNode;
+    renderNextButton: () => ReactNode;
+    onDone: () => void;
+    onSkip: () => void;
+    slides: [];
+    hidePagination: boolean;
+    styles: IntroScreenStyle;
+    showMode: string;
+}
+
+const Touchable: ComponentClass<any> = Platform.OS === "android" ? TouchableNativeFeedback : TouchableOpacity;
+
+export const SwipeableContainer = (props: SwipeableContainerProps): JSX.Element => {
+    const dimensions = Dimensions.get("window");
+    const [width, setWidth] = useState(dimensions.width);
+    const [activeIndex, setActiveIndex] = useState(0);
+    const [flatList, setFlatList] = useState();
+
+    const goToSlide = useCallback(
+        (pageNum: number) => {
+            setActiveIndex(pageNum);
+            if (flatList) {
+                flatList.scrollToOffset({
+                    offset: _rtlSafeIndex(pageNum) * width
+                });
+            }
+        },
+        [width, flatList]
+    );
+
+    const _onNextPress = (): void => {
+        goToSlide(activeIndex + 1);
+        if (props.onSlideChange) {
+            props.onSlideChange(activeIndex + 1, activeIndex);
+        }
+    };
+    const _onPrevPress = (): void => {
+        goToSlide(activeIndex - 1);
+        if (props.onSlideChange) {
+            props.onSlideChange(activeIndex - 1, activeIndex);
+        }
+    };
+
+    const _onPaginationPress = (index: number): void => {
+        const activeIndexBeforeChange = activeIndex;
+        goToSlide(index);
+        if (props.onSlideChange) {
+            props.onSlideChange(index, activeIndexBeforeChange);
+        }
+    };
+
+    const _renderItem = ({ item }: any): JSX.Element => {
+        return <View style={[{ width, flex: 1, alignContent: "stretch" }]}>{item.content}</View>;
+    };
+
+    const _renderDefaultButton = (label: string, bottomStyle: ViewStyle): JSX.Element => {
+        let content = <Text style={props.styles.buttonText}>{label}</Text>;
+        if (props.bottomButton) {
+            content = <View style={[styles.bottomButtonDefault, bottomStyle]}>{content}</View>;
+        }
+        return content;
+    };
+
+    const _renderButton = (
+        label: string,
+        content: () => ReactNode,
+        onPress: () => void,
+        normalButtonStyle: ViewStyle,
+        bottomButtonStyle: ViewStyle
+    ): JSX.Element => {
```

Comment:
Why `JSX.Element` why not `ReactNode`?

Reply:
ReactNode can return undefined and I am defining explicitly the Element